src/masternodes/anchors.cpp

```diff
@@ -526,9 +526,26 @@ void CAnchorIndex::CheckPendingAnchors()
 {
     AssertLockHeld(cs_main);
 
-    std::set<uint256> deletePending;
+    // Get pending anchors and organise by Bitcoin height, anchor height or TX hash
+    auto ByBtcHeight = [](const AnchorRec& a, const AnchorRec& b) {
+        if (a.btcHeight == b.btcHeight) {
+            if (a.anchor.height == b.anchor.height) {
+                return a.txHash < b.txHash;
+            }
+
+            // Higher DeFi height wins
+            return a.anchor.height > b.anchor.height;
+        }
+
+        return a.btcHeight < b.btcHeight;
+    };
```

Comment:
Can we some unittest for this function?

Reply:
@ShengguangXiao We totally can. Will add one tomorrow.